docs/linting/Troubleshooting.md

```diff
@@ -311,6 +311,69 @@ The following rules do not have equivalent checks in TypeScript, so we recommend
 - `import/no-unused-modules`
 - `import/no-deprecated`
 
+#### `import/extensions`
+
+There are two uses for this rule:
+
+1. to enforce file extensions are always used,
+2. to enforce file extensions are never used.
+
+##### Enforcing extensions are used
+
+If you want to enforce file extensions are always used and you're **NOT** using `moduleResolution` `node16` or `nodenext`, then there's not really a good alternative for you, and you should continue using the `import/extensions` lint rule.
+
+If you want to enforce file extensions are always used and you **ARE** using `moduleResolution` `node16` or `nodenext`, then you don't need to use the lint rule at all because TypeScript will automatically enforce that you include extensions (TS 2834 & 2835)!
```

Comment:
Users might not know this refers to specific error codes. And those error codes sometimes change version-to-version. Maybe let's just remove these?

```suggestion
If you want to enforce file extensions are always used and you **ARE** using `moduleResolution` `node16` or `nodenext`, then you don't need to use the lint rule at all because TypeScript will automatically enforce that you include extensions!
```